agenta-backend/agenta_backend/routers/testset_router.py

```diff
@@ -68,11 +85,13 @@ async def upload_file(
                     row_data[columns[i]] = value
                 document["csvdata"].append(row_data)
 
-        result = await testsets.insert_one(document)
+        user = await get_user_object(kwargs["user_id"])
+        testset_instance = TestSetDB(**document, user=user)
+        result = await engine.save(testset_instance)
 
-        if result.acknowledged:
+        if result is not None:
```

Comment:
Why did you remove the ack here? Isn't there a way to see if the operation succeeded

Reply:
There isn't. What I can do is, after saving the instance to the database, I make a new call to retrieve and check if it exists.